src/us/kbase/workspace/kbase/IdentifierUtils.java

```diff
@@ -236,21 +237,11 @@ private static ObjectIdentifier buildObjectIdentifier(
 					+ objcount + ": " + e.getLocalizedMessage(), e);
 		}
 		final boolean searchDAG = longToBoolean(o.getFindReferencePath());
-		if (paths == null && refchain == null && !searchDAG) {
-			return oi;
-		} else if (paths == null) {
-			if (searchDAG) {
-				return new ObjectIDWithRefPath(oi);
-			} else {
-				return new ObjectIDWithRefPath(oi, refchain);
-			}
-		} else {
-			if (searchDAG) {
-				return new ObjIDWithRefPathAndSubset(oi, paths);
-			} else {
-				return new ObjIDWithRefPathAndSubset(oi, refchain, paths);
-			}
-		}
+		return ObjectIdentifier.getBuilder(oi)
+				.withReferencePath(refchain)
+				.withLookupRequired(searchDAG)
+				.withSubsetSelection(paths)
+				.build();
```

Comment:
here we see the true power of this fully armed an operational fluent builder

Reply:
But is the world ready for this mighty builder?

Reply:
It will be... *malevolent chuckle*... it will be.